src/app/service/auth-guard/auth-guard.service.spec.ts

```diff
@@ -0,0 +1,26 @@
+import 'rxjs/add/operator/toPromise';
```

Comment:
I don't see this being used. If you were having errors without it, maybe it should be included in the file where you convert the subscription to a promise.

Reply:
If I don't include that import I get this error:  `Property 'toPromise' does not exist on type 'Observable<Response>'.`

Reply:
I guess you can leave it for now. I could be wrong but I think it probably belongs in a different file. You could check if all your files that use `toPromise` have that import.